File: audiobook/src/main/java/de/ph1b/audiobook/dialog/BaseDialogFragment.java
package de.ph1b.audiobook.dialog;

import android.support.v4.app.DialogFragment;

import de.ph1b.audiobook.utils.BaseApplication;

/**
 * Created by Paul Woitaschek (woitaschek@posteo.de, paul-woitaschek.de) on 04.08.15.
 * Simple base dialog fragment that is only watching for leaks
 */
public class BaseDialogFragment extends DialogFragment {

    @Override
    public void onDestroy() {
        super.onDestroy();

        BaseApplication.leakWatch(this);
    }
}

File: audiobook/src/main/java/de/ph1b/audiobook/misc/BaseDialogFragment.kt
package de.ph1b.audiobook.misc

import android.support.v4.app.DialogFragment

/**
 * Base DialogFragment
 *
 * @author Paul Woitaschek
 */
abstract class BaseDialogFragment : DialogFragment() {

  /** find a callback. The hosting activity must implement [RouterProvider] and the supplied key must match to the instance id of a controller */
  fun <T> findCallback(controllerBundleKey: String): T {
    val routerProvider = activity as RouterProvider
    val router = routerProvider.provideRouter()
    val controllerId: String = arguments.getString(controllerBundleKey)
    @Suppress("UNCHECKED_CAST")
    return router.getControllerWithInstanceId(controllerId) as T
  }
}